apps/api/src/index.ts

```diff
@@ -87,7 +87,18 @@ app.set("trust proxy", true);
 
 app.use(
   cors({
-    origin: config.app.url,
+    origin: function (origin, fn) {
+      if (
+        !origin ||
+        config.app.url === origin ||
+        // Note: the frontend app URL above should already be included in the `corsOrigins` config
+        config.app.corsOrigins.includes(origin)
+      ) {
+        fn(null, true);
+      } else {
+        fn(new ApiError(403, "Forbidden"));
```

Comment:
some ai-gen'd rationale here, but it seems valid:
1. if a "bad" CORS origin makes a request, we were returning a 404 instead of 403
2. if a "bad" CORS origin makes a preflight request, we leaked a _little_ information for the available methods, headers, and credentials values below.

Reply:
@joewagner nice...yeah that's probably easier than making things so explicit